cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/AbstractApplicationDetailFetcher.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.metadata;
+
+import io.cdap.cdap.common.NotFoundException;
+import io.cdap.cdap.proto.ApplicationDetail;
+import io.cdap.cdap.proto.id.ApplicationId;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Interface for fetching {@code ApplicationDetail}
+ */
+public interface AbstractApplicationDetailFetcher {
```

Comment:
Done. Removed "Abstract" 

Yes. But I am a bit leaning toward creating a LocalxxxFetcher that contains an instance of xxxService instead of have xxxService implement the interface, so it looks a bit more symmetric. Coming from C++ background, I find a class implementing multiple different interfaces a bit intuitive.